### DISCUSSION TITLE
Creating playbook rules with regex used keyword field

### DISCUSSION THREAD

**Discussion by btdewey** [Security-Onion-Solutions/securityonion#8283] [General] [answered]
We are adding new plays to playbook.  When we use the sigma "fieldname|re:" function it properly add the regex and doesn't escape special characters, but it uses the fieldname.keyword instead of fieldname.security.  This is causing the plays to not match data as expected.  I'm trying to figure out if this is a case sensitivity issue or something else, but if the regex conversion could use the .security instead of the .keyword subfield that would be helpful.

**Comment by defensivedepth** [comment]
What is the use case for using regex? Regex should be utilized rather sparingly, as it can cause major performance issues.

**Comment by btdewey** [comment]
Should it work or shouldn't it?  Specifically I'm trying to match a Linux
command line switch.   Since there can be characters between the dash and
the option I need to match characters and not spaces or other symbols.
Therefore a wildcard is insufficient.

Bill

**Reply by defensivedepth** [reply]
It should work. Can you share a (sanitized as needed) rule that does not work the way you are expecting?

**Comment by btdewey** [comment]
I was trying to match something simple on the command line field and it
would match with both the base and .security but not .keyword.  sending via
email so may need a code block to display.

process.command_line: /.*dir.*/

**Comment by btdewey** [comment]
We are getting that field from endgame streaming and now I'm wondering if
it actually exists.  I'll check that on Monday and close out the issue if
it's a missing field.

Bill

**Comment by btdewey** [comment, marked as answer]
The ECS compliant field from Endgame registry.path doesn't have a .keyword subfield with our setup.  The Endgame specific field endgame.registry_path does.  We have used the .keyword subfield before for regex so I just didn't think through this issue sufficiently.  We can switch to the endgame field as a work around until we can look into why the subfield isn't there.

**Reply by defensivedepth** [reply]
Sounds good!
